queries.sql

```diff
@@ -1,10 +1,118 @@
+
+/* Begin Transaction B */
+BEGIN;
+
+/* update species col for animals with names ending in mon */
+update animals set species = 'digimon' where name like '%mon';
+
+/* update species col for animals with names not ending in mon */
+update animals set species = 'pokemon' where species IS null;
+
+/* Confirm changes */
+SELECT * from animals;
+
+/* commit changes */
+COMMIT;
+
+/* Verify that change was made and persists after commit */
+SELECT * from animals;
+
+/* Delete all records in the animals table transaction C */
+/* Begin Transaction C */
+BEGIN;
+
+/* delete all records */
+DELETE from animals;
+
+/* verify all records are deleted */
+SELECT * from animals;
+
+/* Rollback changes */
+ROLLBACK;
+
+/* verify all records still exist */
+SELECT * from animals;
+
+/* Begin Transaction D */
+BEGIN;
+
+/* Delete all animals born after Jan 1st, 2022. */
+DELETE from animals
+WHERE date_of_birth > '01-01-2022';
+
+/* create a savepoint */
+SAVEPOINT dob;
+
+/* update all animals' weights multiplied by -1 */
+UPDATE animals
+SET weight_kg =weight_kg * -1;
+
+/* roolback to savepoint */
+ROLLBACK TO dob;
+
+
+/* update all animals' weights that are negative multiplied by -1 */
+UPDATE animals
+SET weight_kg = weight_kg * -1
+WHERE weight_kg < 0;
+
+/* commit the transaction */
+COMMIT;
+
+/* How many animals are there? */
+SELECT COUNT(*) from animals;
+
+/* How many animals have never tried to escape? */
+SELECT COUNT(*) from animals
+WHERE escape_attempts = 0;
+
+/* What is the average weight of animals? */
+SELECT AVG(weight_kg) from animals;
+
+/* Who escapes the most, neutered or not neutered animals? */
+SELECT neutered, count(neutered) as result
+FROM animals
+GROUP BY neutered
+ORDER BY result DESC LIMIT 1;
+
+/* What is the minimum and maximum weight of each type of animal? */
+SELECT species, MAX(weight_kg) AS max_weight, MIN(weight_kg) AS min_weight
+FROM animals
+GROUP BY species;
+
+/* What is the average number of escape attempts per animal type of those born between 1990 and 2000? */
+SELECT species, AVG(escape_attempts) FROM animals
+WHERE date_of_birth BETWEEN '1990-01-01' AND '2000-12-31'
+GROUP BY species;
+
+SELECT * FROM animals JOIN owners ON animals.owner_id=owners.id WHERE owners.full_name='Melody Pond';
+SELECT * FROM animals JOIN species ON animals.species_id=species.id WHERE species.name='Pokemon';
+SELECT * FROM animals RIGHT JOIN owners ON animals.owner_id=owners.id;
+SELECT species.name ,COUNT(*) FROM animals JOIN species ON animals.species_id=species.id GROUP BY species.name;
+SELECT * FROM ((animals JOIN owners ON animals.owner_id=owners.id) JOIN species ON animals.species_id=species.id)
+  WHERE owners.full_name='Jennifer Orwell' AND species.name='Digimon';
+SELECT * FROM animals JOIN owners ON animals.owner_id=owners.id WHERE owners.full_name='Dean Winchester' AND animals.escape_attempts=0;
+SELECT owners.id ,owners.full_name, COUNT(*) FROM animals JOIN owners ON animals.owner_id=owners.id GROUP BY owners.id;
```

Comment:
- [ ] This query, is not definitely correct for listing the `owner` who owns more `animals`. Kindly make sure you implement your query is correct.

Hint: You might say:
```sql
SELECT owners.full_name, COUNT(owners.full_name) FROM animals
   LEFT JOIN owners
   ON animals.species_id = owners.id
   GROUP BY owners.full_name
   ORDER BY COUNT(owners.full_name) DESC
```